be/src/vec/exprs/vcompound_pred.h

```diff
@@ -342,10 +380,9 @@ class VCompoundPred : public VectorizedFnCall {
         return (l_null & r_null) | (r_null & (r_null ^ a)) | (l_null & (l_null ^ b));
     }
 
-    bool _all_child_is_compound_and_not_const() const {
-        return std::ranges::all_of(_children, [](const VExprSPtr& arg) -> bool {
-            return arg->is_compound_predicate() && !arg->is_constant();
-        });
+    bool _all_child_is_not_const() const {
```

Comment:
As we use this method like `!_all_child_is_not_const`, and it is equal to `_has_const_child`, so I think we should implement ``
`_has_const_child`, which is more easy to understand.